R/readGAINS2025final.R

```diff
@@ -0,0 +1,285 @@
+#' Read and combine air pollution emissions, activities and emission factors from GAINS data
+#'
+#' There's no associated convert function, as the disaggregation takes
+#' a combination of subtypes, and it is easier to carry out most
+#' calculations at the GAINS regional level first and then disaggregate
+#' the results
+#'
+#' @return  Activity levels, emissions or emission factors at the level of
+#'          25 GAINS regions, 35 GAINS sectors and 7 species:
+#'          magclass object with dimensions region, year, and ssp.scenario.sectorGAINS.species
+#' @author Gabriel Abrahao, Laurin Koehler-Schindler
+#' @param subtype "emifacs", "emissions", "activities"
+#' @importFrom stats na.omit
+#' @importFrom utils timestamp globalVariables
+#' @importFrom tidyr complete
+
+utils::globalVariables(c(
+  "EMF30_AGG", "EMF30_DET", "EMF30_MIXED", "Group_Region", "POLLUTANT_FRACTION",
+  "SCENARIO", "VARIABLE", "VARIANT", "region", "scen", "scenario",
+  "sectorGAINS", "species", "ssp", "value", "vartype", "year"
+))
+
+
+readGAINS2025final <- function(subtype) {
+  # ================================================================================================
+  # SECTOR INFORMATION =============================================================================
+  # ================================================================================================
+  sectors <- read.csv("GAINS2025_sectors_2025-07-02.csv")
+  sectors_agg <- na.omit(unique(sectors$EMF30_AGG))
+  # Aggregated GAINS sectors do not contain Waste, thus these selected sectors are added from the
+  # detailed GAINS sectors
+  sectors_det_selected <- c(
+    "Waste_Solid_Municipal",
+    "Waste_Water_Municipal"
+  )
+  # According to email "ScenarioMIP air pollution EFs - CORRECTION to Transformation"
+  # from Zig (GAINS team) on July 4, the following sectors need to be corrected
+  # for historical and ScenarioMIP emission factors
+  sectors_to_correct <- c(
```

Comment:
personally, I am not a fan of declaring stuff in the very beginning, unless you use it many times. in this case, `sectors_to_correct` is only used in `correctSectors`, so why not declare it right before? makes reading your code easier, as i do not have to browse back and forth to understand it